security/pkg/pki/util/generate_csr_test.go

```diff
@@ -25,80 +26,126 @@ import (
 
 func TestGenCSR(t *testing.T) {
 	// Options to generate a CSR.
-	csrOptions := CertOptions{
-		Host:       "test_ca.com",
-		Org:        "MyOrg",
-		RSAKeySize: 512,
+	cases := map[string]struct {
+		csrOptions CertOptions
+	}{
+		"GenCSR with RSA": {
+			csrOptions: CertOptions{
+				Host:       "test_ca.com",
+				Org:        "MyOrg",
+				RSAKeySize: 2048,
+			},
+		},
+		"GenCSR with EC": {
+			csrOptions: CertOptions{
+				Host: "test_ca.com",
+				Org:  "MyOrg",
+				IsEC: true,
+			},
+		},
 	}
 
-	csrPem, _, err := GenCSR(csrOptions)
+	for id, tc := range cases {
+		csrPem, _, err := GenCSR(tc.csrOptions)
 
-	if err != nil {
-		t.Errorf("failed to gen CSR")
-	}
+		if err != nil {
```

Comment:
But: Extra spacing

Reply:
It's now part of a table test, hence the spacing.